flink-runtime/src/main/scala/org/apache/flink/runtime/taskmanager/TaskManager.scala

```diff
@@ -411,6 +411,23 @@ class TaskManager(
             log.debug(s"Cannot find task to fail for execution ${executionID})")
           }
 
+        // stops a task
+        case StopTask(executionID) =>
+          val task = runningTasks.get(executionID)
+          if (task != null) {
+            try {
+              task.stopExecution()
+              sender ! new TaskOperationResult(executionID, true)
+            } catch {
+              case t: Throwable =>
+                        sender ! new TaskOperationResult(executionID, false, t.getMessage())
```

Comment:
I thought I need to send back a `TaskOperationResult` (which does not take only `String` as parameter for an error message). And `stopExecution` might throw an `UnsupportedOperationException` (which should not happen, if code is correct). Please give some more advice (I am not to much familiar with Akka)